Branch-SDK/Branch-SDK/BranchShareLink.m

```diff
@@ -0,0 +1,253 @@
+//
+//  BranchShareLink.m
+//  Branch-SDK
+//
+//  Created by Edward Smith on 3/13/17.
+//  Copyright © 2017 Branch Metrics. All rights reserved.
+//
+
+#import "BranchShareLink.h"
+#import "BranchConstants.h"
+#import "BNCFabricAnswers.h"
+#import "BranchActivityItemProvider.h"
+#import "BNCDeviceInfo.h"
+@class BranchShareActivityItem;
+
+typedef NS_ENUM(NSInteger, BranchShareActivityItemType) {
```

Comment:
I don't like to expose enums or types unless there is a good reason.

Once the type is exposed we have to support in the future.